### ISSUE TITLE
Synchronize GitHub teams with GSuite

### ISSUE THREAD

**Issue by Ivan-Brigidano** [thewca/worldcubeassociation.org#4705]
I don't know if this would be even possible. But if so, it would be much easier to maintain for administrators.

**Comment by Ivan-Brigidano** [comment]
According to #449 Slack isn't possible because of the current plan. But maybe GitHub?

**Comment by dunkOnIT** [comment]
Asking from a place of pure ignorance: 
1. Can we clarify what sort of synchronisation is being proposed here? 
2. Can we quantify how much benefit this would create for administrators? 

My assumption is that WCA subcommittee memberships change relatively infrequently, and the associated admin isn't particularly onerous. While I'm sure it isn't anyone's definition of fun, I'd expect it to be 5-10 minutes of setup per user at max, a few times a year? 

Understanding the above would help us prioritise this issue, or backlog it, as necessary.

**Comment by gregorbg** [comment]
There are UIs on the non-public (Staff-only) part of the WCA website that allow to manage membership of the different Teams and Committees. These memberships are then automatically synced to Google Workspace, such that mailing lists like software@worldcubeassociation.org always reach the correct recipients without having to manually add people in the Google Workspace admin panel as well.

It's pretty much just convenience and your summary in terms of workload vs. benefit is accurate. However, consider that there are a lot of different WCA teams that can sum up the workload. This issue is mostly useful for people who want to get a better understanding of the codebase, because this implementation strikes the balance between "super-trivial one-line change" and "complex project". Hence the "good second issue" label

**Comment by dunkOnIT** [comment]
A cursory glance at the Github documentation suggests that adding users to the WCA Github Organisation would be possible, and given that it presents good learning potential it makes sense to run with it.

Given that there's quite a bit of (now unrelated) discussion in this issue, I suggest the following:
* Create an issue called "Add users to Github organisation/team from WCA membership pannel"
* Label it appropriately
* Close this issue in favour of the new one, with links going both ways

I favour this approach because its easier for a naive collaborator to see what needs to be done. I see now that I can also just edit the title and original post of this issue, but that feels a bit weird - let me know what you think.

**Comment by gregorbg** [comment]
On second thought, this might be not so trivial after all, because GitHub users are not always identified by the WCA mail address. We can do automated GSuite syncs based only on the assumption that Staff members add their WCA email address to their WCA website profile, and we would have to introduce (and store in the DB, which makes this not so trivial) a second field for "GitHub username". Because even if APIs exist for this task, our code wouldn't know that GitHub user `gregorbg` is indeed `2012BILL01`.

**Comment by dunkOnIT** [comment]
Agreed - I had in mind that adding someone to Github org would be optional, only triggered if a Github email/username is entered. Wasn't aware that database interactions increase complexity significantly - that's good to know. 

One way of resolving this - which I would argue against - would be creating a new Github user using the WCA email. Reason for arguing against this is the degree to which a Github profile acts as a portfolio for many. 

Another option could be to split it into two tasks: 
- [ ] Create frontend form/component (relatively easy/good second issue material)
- [ ] Implement database call that stores the Github details/returns any confirmations needed

Seems difficult to coordinate in practice though, and I'm not sure if the codebase is loosely coupled enough for that to be viable?

**Comment by dunkOnIT** [comment]
Seems easier to manage GIthub access separately than add database fields which track Github username? I guess this would allow users to specify their own Github username instead of whoever is doing the administration asking them for it. But still, this doesn't seem particularly beneficial
